app/services/rules_cleanup_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# This class cleans up unused rules and reimports
+class RulesCleanupService
+  class << self
+    def run!
+      # Clean up rules that don't belong to any profile
+      Rule.where.not(id: ProfileRule.pluck(:rule_id)).delete_all
+
+      # Clean up rules that belong only to canonical profiles
+      delete_rules(select_rules)
+
+      # Reimport all rules if we're not in the testing environment
+      Rake::Task['ssg:import_rhel_supported'].execute unless Rails.env.test?
+    end
+
+    def select_rules
+      profiles = Profile.canonical(false).select(:parent_profile_id).distinct
+      used_rules = ProfileRule.select(:rule_id).where(profile_id: profiles)
+
+      Rule.where.not(id: RuleResult.select(:rule_id))
+          .where.not(id: used_rules).select(:id)
+    end
+
+    def delete_rules(rules)
+      RuleReferencesRule.where(rule_id: rules).delete_all
+      RuleReference.joins(:rules).where('rules.id' => nil).delete_all
```

Comment:
When `rules.id` will be `NULL` in the case of an `INNER JOIN`?

Reply:
I thought Rails sorts this out OOTB, but better safe than sorry :see_no_evil: fixing